### PR TITLE
Improvements to PhantomData<T>'s docs 👻

### LINKED ISSUE #22914: PhantomData is incomprehensible
I need to give a Rust slice to a C API that will return an opaque handle (that actually contains a pointer to the slice), and the slice must outlive the handle.

`rustc` nicely suggested using `PhantomData` for this, but the description of phantom data (and its RFC) for me is so far removed from my mental model/terminology of the problem, that it [sounds like git man pages to me](http://git-man-page-generator.lokaltog.net/). 

* "…tell the compiler about fake fields" — *fake field* doesn't sound like a thing I want to create. I'm not sure if a fake field is an actual Rust concept. I'm hoping it's a different way of saying a "pointer that Rust isn't keeping track of".

* "inform the compiler that one or more instances of the type T could be dropped" — but does it actually drop anything? I want to prevent value from being dropped, so this confuses me.

The RFC is even worse, since it uses type variance terminology which is so confusing, that the most comprehensible description of them on Wikipedia is a **3D cube**!

3d variance

### PR DESCRIPTION
**Pull request by steveklabnik**
Fixes #22914

Said issue was mostly fixed, as there wasn't any examples when it was initially posted. This is mostly just some re-wording of some things and some cleanup

### CODE REVIEW

File: src/libcore/marker.rs [hunk 1]
```diff
@@ -346,17 +346,16 @@ impl<T:?Sized> MarkerTrait for T { }
...
-/// `PhantomData` is a way to tell the compiler about fake fields.
-/// Phantom data is required whenever type parameters are not used.
-/// The idea is that if the compiler encounters a `PhantomData<T>`
-/// instance, it will behave *as if* an instance of the type `T` were
-/// present for the purpose of various automatic analyses.
+/// `PhantomData<T>` allows you to describe that a type acts as if it stores a value of type `T`,
+/// even though it does not. This allows you to inform the compiler about certain safety properties
+/// of your code.
+///
+/// Though they both have scary names, `PhantomData<T>` and "phantom types" are unrelated. 👻👻👻
```
Comment (Gankro) [inline src/libcore/marker.rs]: Weird unicode junk at the end of this line?

Comment (steveklabnik) [inline src/libcore/marker.rs]: it's not weird junk! it's ghosts!

Comment (Gankro) [inline src/libcore/marker.rs]: Ahaha. Doesn't render in my browser, but :+1:

Comment (steveklabnik) [inline src/libcore/marker.rs]: :ghost: :ghost: :ghost:

Comment (steveklabnik) [inline src/libcore/marker.rs]: (if you're on linux, i installed `ttf-ancient-fonts` to get emoji to render)

**Pull request merged**